src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/AuthController.java

```diff
@@ -45,8 +48,21 @@ public ResponseEntity<Void> registerUser(@Valid @RequestBody AuthRegisterDTO aut
     }
 
     @PostMapping("/refresh")
-    public ResponseEntity<AuthAccessTokenDTO> refreshToken(@CookieValue(name = "refresh_token") String refreshToken) {
-        //TODO
-        return new ResponseEntity<>(new AuthAccessTokenDTO("TODO-accessTOKEN"), HttpStatus.OK);
+    public ResponseEntity<AuthAccessTokenDTO> refreshToken(@CookieValue(name = "refresh_token", required = false) String refreshToken, HttpServletRequest request, HttpServletResponse response) {
+        String accessToken = request.getHeader("Authorization").replace("Bearer ", "");
+
+        if (userService.validateToken(accessToken) && userService.validateToken(refreshToken)) {
```

Comment:
access token should have valid signature but can be expired in that case
on the other hand refresh token should have valid signature as well as cannot be expired